### PR TITLE
#166789992 User should be able to sign out

### PR DESCRIPTION
**Pull request by niyoceles**
#### What does this PR do?
- Enable user sign out
#### Description of Task to be completed
- Configure passport to clean session of the user

#### How should this be manually tested?
1. Run the following commands in Terminal
```
git clone git@github.com:andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-user-sign-out-166789992
```
2. run `npm install`

3. Install your local database in Postgres

4. Rename `.env.sample` to `.env` Open it and replace  `database-name` your created database

5. run `npm run start`

6. Sign out to this endpoints
- `http://localhost:3000/api/v1/auth/signout`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166789992](#166789992)
#### Screenshots (if appropriate)
N/A

#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
eslint throws a trailing commas warning, kindly fix them

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]
The fact that hound is having issue means that there is something wrong here. Please I think you should please obey all eslint rules

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,19 @@
+env:
+  global:
+    - NODE_ENV=test
+    - secret = authorshaven
+    - DATABASE_TEST=postgres://postgres:admin5423/users_test
```
Comment (kodek-sleuth) [inline .travis.yml]: I am having a security issue here, you **should not** pass our secret and DB_URL in this script

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
 services:
   - postgresql
+before_script:
+  - psql -c 'CREATE DATABASE users_test;' -U postgres
+  - npm i sequelize-cli -g
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
 after_success:
-  - npm test
```
Comment (kodek-sleuth) [inline .travis.yml]: The presence of this line theoretically is incorrect in all manner, after_success means the command travis will run after running our script. Please we should only have one command here

File: controllers/welcomeController.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import dotenv from "dotenv";
-import db from "../models/databaseUrl";
+import dotenv from 'dotenv';
+
 dotenv.config();
 
 class WelcomeController {
   static welcome(req, res) {
     res.status(200).json({
-      msg: "Welcome to Authors Haven API, a #92Explorers Product."
```
Comment (kodek-sleuth) [inline controllers/welcomeController.js:8]: since you wrote this class, I prefer you please use the full form of the word message

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

**Review by kodek-sleuth — COMMENTED** [review]
I think in the PR description, we deserve to know what happens to the token after the user has signed out

**Review by Hervera — CHANGES_REQUESTED** [review]
Make the messages in tests be understandable, like 
`it('Not found Url should return 404', (done) => {` 
to
 `it('Should return 404 if url is not found ', (done) => {`

**Comment by niyoceles** [comment]
> I think in the PR description, we deserve to know what happens to the token after the user has signed out

It has a method which  will remove the `req.user` property and clear the login session

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]
I pretty certain this feature did not require all 3 commits, so please squash your commits to one

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,20 @@
+env:
+  global:
+    - NODE_ENV=test
+    - secret = authorshaven
+    - DATABASE_TEST=postgres://postgres:admin@localhost/users_test
```
Comment (kodek-sleuth) [inline .travis.yml:5]: We talked about this and decided that this is the travis.yml script that we will be using
Screen Shot 2019-07-09 at 08 55 09

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

File: config/passport.js [hunk 5]
```diff
@@ -1 +1,29 @@
-import passport from "./node_modules/passport";
```
Comment (kodek-sleuth) [inline config/passport.js:1]: When importing an installed module, you don't need to necessary include node_modules

File: controllers/socialUserController.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            // token,
+            firstName: req.user.name.givenName,
+            lastName: req.user.name.familyName,
+            email: req.user.emails[0].value,
+            provider: req.user.provider
+          }
+        });
+      }
+    } catch (error) {
+      return res.status(500).json({
```
Comment (kodek-sleuth) [inline controllers/socialUserController.js:29]: Improve on this error ro be a little specific by testing the errors encountered in the try and manifesting on them in the catch

File: migrations/20190707101657-create-user.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
```
Comment (kodek-sleuth) [inline migrations/20190707101657-create-user.js:36]: Try and implement es6 standards like this
Screen Shot 2019-07-09 at 08 59 17

File: models/User.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+export default (sequelize, DataTypes) => {
+  const User = sequelize.define(
+    'User',
```
Comment (kodek-sleuth) [inline models/User.js:3]: This is how the model should be structured for purposes of work efficiency as the teammates have concluded
Screen Shot 2019-07-09 at 09 01 37

**Review by kodek-sleuth — APPROVED** [review]

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]
Implement es6 as much as possible

File: config/passport.js (hunk 5 above)
Comment (elemanhillary-zz) [reply config/passport.js:1]: that line of code was already there in the previous original source code, am padoning him on that, but atleast he should have removed it

File: controllers/socialUserController.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
+import models from '../models';
+
+export default class User {
+  static async userFacebookGoogle(req, res) {
+    // console.log('in controller =====', req.user);
+    try {
+      const facebookGoogleUser = await models.User.findOrCreate({
+        where: { email: req.user.emails[0].value },
```
Comment (elemanhillary-zz) [inline controllers/socialUserController.js:8]: kindly use destructuring on the `req.user` object

File: controllers/socialUserController.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+      if (req.user.displayName) {
+        const [firstName, lastName] = req.user.displayName.split(' ');
+        userInfo.firstName = firstName;
+        userInfo.lastName = lastName;
+      }
+
+      const twitterUser = await models.User.findOrCreate({
+        where: { username: req.user.username, provider: req.user.provider },
+        defaults: {
+          firstName: userInfo.firstName,
```
Comment (elemanhillary-zz) [inline controllers/socialUserController.js:48]: kindly use destructuring on `req.user` to obtain its keys

File: controllers/welcomeController.js [hunk 11]
```diff
@@ -1,11 +1,11 @@
...
+      msg: 'Welcome to Authors Haven API, a #92Explorers Product.'
```
Comment (elemanhillary-zz) [inline controllers/welcomeController.js:8]: rename `msg` to `message`

File: migrations/20190707101657-create-user.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    return queryInterface.dropTable('Users');
+  }
+};
```
Comment (elemanhillary-zz) [inline migrations/20190707101657-create-user.js:39]: Add a new line on the EOF

File: models/User.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+    {
+      firstName: DataTypes.STRING,
+      lastName: DataTypes.STRING,
+      email: DataTypes.STRING,
+      username: DataTypes.STRING,
+      provider: DataTypes.STRING
```
Comment (elemanhillary-zz) [inline models/User.js:9]: missing trailing comma dangle, kindly fix this

File: models/User.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    {}
+  );
+  User.associate = function (models) {
```
Comment (elemanhillary-zz) [inline models/User.js:13]: line `13` shouldn't be there since its unused

File: swaggerSetUp/ah-92explorers-api.js [hunk 15]
```diff
@@ -1,28 +1,28 @@
...
 
 const swaggerDefinition = {
   info: {
-    title: "Authors Haven",
-    version: "1.0.0",
+    title: 'Authors Haven',
+    version: '1.0.0',
     description:
-      "Create a community of like minded authors to foster inspiration and innovation by leveraging the modern web"
+      'Create a community of like minded authors to foster inspiration and innovation by leveraging the modern web'
```
Comment (elemanhillary-zz) [inline swaggerSetUp/ah-92explorers-api.js:8]: fix all missing trailing comma-dangle

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
Kindly implement and fix all issues i have raised in the review comment

**Pull request closed without merging**